main.cpp

```diff
@@ -458,7 +458,7 @@ void RemoteControl::switchMode(bool hdmi)
 
 int RemoteControl::turnOff(bool isButtonDown)
 {
-    if (onOffSwitch == 1 && isButtonDown == true)
+    if (onOffSwitch && isButtonDown)
```

Comment:
`int onOffSwitch = 1;`
You were previously checking if `onOffSwitch == 1` which is a valid comparison.
by removing the `== 1`, you are effectively writing `if( onOffSwitch equals any number other than 0` but without being explicit about it.  By not being explicit about the comparison, you have introduced the potential for a bug.
- [ ] revise